### PR TITLE
swayfx: update to 0.4

### PR DESCRIPTION
**Pull request by shmishtopher**
closes #51020

#### Testing the changes
- I tested the changes in this PR: **briefly**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, x86_64-glibc

### CODE REVIEW

File: srcpkgs/scenefx/template [hunk 1]
```diff
@@ -0,0 +1,21 @@
+# Template file for 'scenefx'
+pkgname=scenefx
+version=0.1
+revision=1
+archs="i686 x86_64"
```
Comment (classabbyamp) [inline srcpkgs/scenefx/template:5]: why?

File: srcpkgs/scenefx/template [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+build_style=meson
+configure_args="-Dwerror=false -Db_ndebug=false"
+hostmakedepends="pkg-config wayland-devel scdoc"
+makedepends="wlroots0.17-devel"
+depends="libdrm pixman"
+short_desc="Drop-in replacement for the wlroots scene API"
+maintainer="Christopher K. "Shmish" Schmitt <me@shmish.dev>"
```
Comment (classabbyamp) [inline srcpkgs/scenefx/template:12]: needs to be single-quotes for the nickname

File: srcpkgs/scenefx/template (hunk 1 above)
Comment (shmishtopher) [reply srcpkgs/scenefx/template]: Whoops, looks like that slipped through the `xnew` template.  Corrected now.

File: srcpkgs/scenefx/template (hunk 2 above)
Comment (shmishtopher) [reply srcpkgs/scenefx/template]: Noted and corrected.

File: srcpkgs/scenefx/template [hunk 3]
```diff
@@ -0,0 +1,20 @@
+# Template file for 'scenefx'
+pkgname=scenefx
+version=0.1
+revision=1
+build_style=meson
+configure_args="-Dwerror=false -Db_ndebug=false"
+hostmakedepends="pkg-config wayland-devel scdoc"
+makedepends="wlroots0.17-devel"
+depends="libdrm pixman"
```
Comment (classabbyamp) [inline srcpkgs/scenefx/template:9]:
```suggestion
```
autodetected

File: srcpkgs/scenefx/template [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+maintainer="Christopher K. 'Shmish' Schmitt <me@shmish.dev>"
+license="MIT"
+homepage="https://github.com/wlrfx/scenefx"
+changelog="https://github.com/wlrfx/scenefx/releases/"
+distfiles="https://github.com/wlrfx/scenefx/archive/refs/tags/${version}.tar.gz"
+checksum=f5c889f4c826a4216628bf1e7e48071fc33e7774b5e3d51e6fee6e571e420827
+
+post_install() {
+	vlicense LICENSE
+}
```
Comment (classabbyamp) [inline srcpkgs/scenefx/template:20]: needs a -devel subpackage

File: srcpkgs/scenefx/template (hunk 3 above)
Comment (shmishtopher) [reply srcpkgs/scenefx/template]: fixed, xbps is very cool

File: srcpkgs/scenefx/template (hunk 4 above)
Comment (shmishtopher) [reply srcpkgs/scenefx/template:19]: added, and `swayfx` package modified to use `-devel` dependency

**Pull request merged**